components/public-api-server/pkg/apiv1/workspace.go

```diff
@@ -290,6 +290,26 @@ func (s *WorkspaceService) DeleteWorkspace(ctx context.Context, req *connect.Req
 	return connect.NewResponse(&v1.DeleteWorkspaceResponse{}), nil
 }
 
+func (s *WorkspaceService) UpdateWorkspaceUserPin(ctx context.Context, req *connect.Request[v1.UpdateWorkspaceUserPinRequest]) (*connect.Response[v1.UpdateWorkspaceUserPinResponse], error) {
```

Comment:
What if this was just a property on the `Workspace` and it was pinned by sending `UpdateWorkspace({ id: 123, pinned: true })`?

Reply:
The reason for this is we should really try not to have an RPC for every property a workspace can have.

Reply:
I agree with pinned being just a boolean in the request, but wanted to keep it consistent with server API. Can change it

Reply:
From discussions, we were hoping to move away from the pattern that server establishes (rpc per property) into more CRUD based ones (which happen to be easier to use by customers also).